streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -248,7 +252,7 @@ public synchronized State state() {
         return Collections.unmodifiableMap(metrics.metrics());
     }
 
-    private class StreamStateListener implements StreamThread.StateListener {
+    private final class StreamStateListener implements StreamThread.StateListener {
         @Override
         public synchronized void onChange(final StreamThread thread,
```

Comment:
Thanks @dguy. That makes sense. @original-brownbear, maybe you can do a follow-up that does that.